glean-core/src/upload/directory.rs

```diff
@@ -0,0 +1,342 @@
+// This Source Code Form is subject to the terms of the Mozilla Public
+// License, v. 2.0. If a copy of the MPL was not distributed with this
+// file, You can obtain one at https://mozilla.org/MPL/2.0/.
+
+//! Pings directory processing utilities.
+
+use std::cmp::Ordering;
+use std::fs::{self, File};
+use std::io::{BufRead, BufReader};
+use std::path::{Path, PathBuf};
+use std::str::FromStr;
+
+use log;
+use serde_json::Value as JsonValue;
+use uuid::Uuid;
+
+use super::PingRequest;
+use crate::Result;
+
+/// Get the file name from a path as a &str.
+///
+/// # Panics
+///
+/// Won't panic if not able to get file name.
+fn get_file_name_as_str(path: &Path) -> Option<&str> {
+    match path.file_name() {
+        None => {
+            log::warn!("Error getting file name from path: {}", path.display());
+            None
+        }
+        Some(file_name) => {
+            let file_name = file_name.to_str();
+            if file_name.is_none() {
+                log::warn!("File name is not valid unicode: {}", path.display());
+            }
+            file_name
+        }
+    }
+}
+
+/// Manages the pending pings directories.
+#[derive(Debug)]
+pub struct PingDirectoryManager {
+    /// Path to the pending pings directory.
+    data_path: PathBuf,
+}
+
+impl PingDirectoryManager {
+    /// Creates a new directory manager.
+    ///
+    /// # Arguments
+    ///
+    /// * `data_path` - Path to the pending pings directory.
+    ///
+    /// # Panics
+    ///
+    /// Will panic if unable to parse the data_path.
+    pub fn new(data_path: &str) -> Self {
+        Self {
+            data_path: PathBuf::from_str(data_path).expect("data_path must be a valid path."),
+        }
+    }
+
+    /// Get the pending pings directory path.
+    pub fn get_dir(&self) -> PathBuf {
+        self.data_path.join("pending_pings")
+    }
+
+    /// Attempts to delete a ping file.
+    ///
+    /// ## Arguments
+    ///
+    /// * `uuid` - The UUID of the ping file to be processed
+    ///
+    /// ## Panics
+    ///
+    /// Won't panic if unable to delete the file.
+    pub fn delete_file(&self, uuid: &str) {
```

Comment:
Having the `data_path` saved, allowed me to do things such as only pass the `uuid` to the `delete_file` and `process_file` functions.

Apart from that, I only changed a little bit the structure of the docstrings here to match the structure of it in the other files and adapted the tests to this new structure.